### PR TITLE
[Remove Stream/Optional usage] LPS-178729 module headless-commerce-admin-catalog-impl part 2

### PR DESCRIPTION
**Pull request by JanisZhang**

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:test:headless

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java [hunk 1]
```diff
@@ -472,11 +467,9 @@ private CPDefinition _addOrUpdateProduct(Product product) throws Exception {
 
 		serviceContext.setAssetTagNames(assetTagNames);
 
-		Map<String, Serializable> expandoBridgeAttributes =
-			_getExpandoBridgeAttributes(product);
-
-		if (expandoBridgeAttributes != null) {
-			serviceContext.setExpandoBridgeAttributes(expandoBridgeAttributes);
+		if (_getExpandoBridgeAttributes(product) != null) {
```
Comment (dantewang) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:470]: @JanisZhang The commit says "won't return null" but the null check is still here. Did you forget to remove it?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java [hunk 2]
```diff
@@ -500,16 +493,9 @@ private CPDefinition _addOrUpdateProduct(Product product) throws Exception {
...
-
-			for (Category category : categories) {
-				if (category.getId() != null) {
-					assetCategoryIds.add(category.getId());
-				}
-			}
-
 			serviceContext.setAssetCategoryIds(
-				ArrayUtil.toLongArray(assetCategoryIds));
+				transformToLongArray(
```
Comment (dantewang) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:497]: @JanisZhang We are going to revert the change that adds this method to the base classes in RestBuilder. Please resend without this change.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java (hunk 1 above)
Comment (JanisZhang) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:470]: @dantewang SORRY.

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java (hunk 2 above)
Comment (JanisZhang) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/resource/v1_0/ProductResourceImpl.java:497]: @dantewang OK.

**Comment by JanisZhang** [comment]
ci:stop
